include/deal.II/fe/fe_dgq.h

```diff
@@ -460,6 +460,12 @@ class FE_DGQLegendre : public FE_DGQ<dim, spacedim>
    */
   FE_DGQLegendre(const unsigned int degree);
 
+  /**
+   * Constructor for tensor product polynomials based on the
+   * given polynomials @p poly.
+   */
+  FE_DGQLegendre(const std::vector<Polynomials::Polynomial<double>> &poly);
```

Comment:
I don't like this constructor because it will yield a polynomial space that is not a Legendre space if the given general polynomial space is not a 1D Legendre space. I think we should rename this class into `FE_DGQModal` or similar (or `FE_DGQArbitraryPolynomialSpace`, because we are not tied to use a modal space, either, if we directly hand in the polynomials), implement all functionality there including the constructor you use here, and set up `FE_DGQLegendre` as a light-weight class on top of `FE_DGQModal`.

Also note the comments I gave in the other class: It is the user's responsibility to provide useful a useful polynomial space.